File: android/app/src/main/java/ae/findmybay/data/repo/PaymentRepository.kt
package ae.findmybay.data.repo

import ae.findmybay.data.api.PaymentApi
import ae.findmybay.data.api.dto.PaymentIntentDto
import ae.findmybay.data.api.dto.PaymentStatusDto
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class PaymentRepository @Inject constructor(
    private val api: PaymentApi,
) {
    suspend fun createIntent(bookingId: String): PaymentIntentDto = api.createIntent(bookingId)

    suspend fun status(paymentRef: String): PaymentStatusDto = api.status(paymentRef)
}

File: ios/FindMyBay/Data/Repo/PaymentRepository.swift
// PaymentRepository.swift
//
// Thin pass-through over `PaymentAPI`. Payments returns DTOs directly to the
// view-models because the wire shape is already user-facing (hostedPageUrl,
// processor, status). Mirrors `PaymentRepository.kt`.

import Foundation

actor PaymentRepository {
    private let api: PaymentAPI

    init(api: PaymentAPI) {
        self.api = api
    }

    func createIntent(bookingId: String) async throws -> PaymentIntentDTO {
        try await api.createIntent(bookingId: bookingId)
    }

    func status(paymentRef: String) async throws -> PaymentStatusDTO {
        try await api.status(ref: paymentRef)
    }
}
